node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/HostCapacityMaintainer.java

```diff
@@ -219,15 +220,19 @@ private List<Node> provisionHosts(int count, NodeResources nodeResources, Option
             }
             Version osVersion = nodeRepository().osVersions().targetFor(NodeType.host).orElse(Version.emptyVersion);
             List<Integer> provisionIndices = nodeRepository().database().readProvisionIndices(count);
+            HostSharing sharingMode = clusterType.map(ClusterSpec.Type::isContainer).orElse(false) ? HostSharing.exclusive : HostSharing.shared;
```

Comment:
It would be nice to reuse Preparer::hostSharing, but with HostSharing.shared instead of HostSharing.any.  (:thinking: I wonder if it is really necessary to have both `any` and `shared`?)
For instance, we want HostSharing.shared for public containers in test and staging.  And HostSharing.shared in main systems.